mm2src/coins/eth.rs

```diff
@@ -6179,3 +6197,168 @@ async fn get_eth_gas_details(
         },
     }
 }
+
+impl ToBytes for Signature {
+    fn to_bytes(&self) -> Vec<u8> { self.to_vec() }
+}
+
+impl ToBytes for SignedEthTx {
+    fn to_bytes(&self) -> Vec<u8> {
+        let mut stream = RlpStream::new();
+        self.rlp_append(&mut stream);
+        // Handle potential panicking.
+        if stream.is_finished() {
+            Vec::from(stream.out())
+        } else {
+            // TODO: Consider returning Result<Vec<u8>, Error> in future refactoring for better error handling.
+            warn!("RlpStream was not finished; returning an empty Vec as a fail-safe.");
+            vec![]
+        }
+    }
+}
+
+#[derive(Debug, Display)]
+pub enum CoinAssocTypesError {
+    InvalidHexString(String),
+    TxParseError(String),
+    ParseSignatureError(String),
+    KeysError(keys::Error),
+}
+
+impl From<DecoderError> for CoinAssocTypesError {
+    fn from(e: DecoderError) -> Self { CoinAssocTypesError::TxParseError(e.to_string()) }
+}
+
+impl From<keys::Error> for CoinAssocTypesError {
+    fn from(e: keys::Error) -> Self { CoinAssocTypesError::KeysError(e) }
+}
+
+#[derive(Debug, Display)]
+pub enum NftAssocTypesError {
+    Utf8Error(String),
+    ParseContractTypeError(ParseContractTypeError),
+    ParseTokenContractError(String),
+}
+
+impl From<ParseContractTypeError> for NftAssocTypesError {
+    fn from(e: ParseContractTypeError) -> Self { NftAssocTypesError::ParseContractTypeError(e) }
+}
+
+impl CoinAssocTypes for EthCoin {
+    type Address = Address;
+    type AddressParseError = MmError<CoinAssocTypesError>;
+    type Pubkey = HtlcPubKey;
+    type PubkeyParseError = MmError<CoinAssocTypesError>;
+    type Tx = SignedEthTx;
+    type TxParseError = MmError<CoinAssocTypesError>;
+    type Preimage = SignedEthTx;
+    type PreimageParseError = MmError<CoinAssocTypesError>;
+    type Sig = Signature;
+    type SigParseError = MmError<CoinAssocTypesError>;
+
+    fn my_addr(&self) -> &Self::Address { &self.my_address }
```

Comment:
I can do it in my PR

Reply:
My comment was not related to naming :)
It's because `my_address` field is removed and `derivation_method` is used instead in EVM HD wallet PR.

Reply:
ref. https://github.com/KomodoPlatform/komodo-defi-framework/blob/26cca4beb06c30e8a4534866d6763cd493dc66a3/mm2src/coins/eth.rs#L451-L455

Reply:
> My comment was not related to naming :)
> 
> It's because `my_address` field is removed and `derivation_method` is used instead in EVM HD wallet PR.

Ohh, I decided its related to dimxy question about naming, will delete commit. 

ps: I can keep if you like v2 name more.

Reply:
We can keep the original name since this not really a v2

Reply:
For more context, this is how this part has been changed for utxo in the HD wallet PRs https://github.com/KomodoPlatform/komodo-defi-framework/blob/26cca4beb06c30e8a4534866d6763cd493dc66a3/mm2src/coins/utxo.rs#L1056-L1066
Still not handled properly but I didn't want to handle errors in swap state machine code in multiple places related to `my_addr` function. But I will need to fix it either way.